src/provider/provider_coarse.c

```diff
@@ -175,6 +437,8 @@ typedef struct coarse_memory_provider_t {
 
     block_t *block_list;
     block_t *alloc_list;
+    struct ravl *free_blocks;
```

Comment:
Add appropriate checks to debug_check() function. Eg check the number of free blocks. allocated blocks, all pointers etc.

Reply:
Done